cmd/list.go

```diff
@@ -333,17 +294,128 @@ func (cooked cookedListCmdArgs) HandleListContainerCommand() (err error) {
 	}
 
 	if cooked.RunningTally {
-		glcm.Info("")
-		glcm.Info("File count: " + strconv.Itoa(int(fileCount)))
+		ls := cooked.newListSummary(fileCount, sizeCount)
+		response := AzCopyResponse[AzCopyListSummary]{ResponseType: "AzCopyListSummary", ResponseValue: ls}
+		glcm.Output(func(format common.OutputFormat) string {
+			if format == common.EOutputFormat.Json() {
+				jsonOutput, err := json.Marshal(response)
+				common.PanicIfErr(err)
+				return string(jsonOutput)
+			} else {
+				return ls.String()
+			}
+		})
+	}
 
-		if cooked.MachineReadable {
-			glcm.Info("Total file size: " + strconv.Itoa(int(sizeCount)))
-		} else {
-			glcm.Info("Total file size: " + byteSizeToString(sizeCount))
+	return nil
+}
+
+type AzCopyResponse[T any] struct {
```

Comment:
This is *close* to the original image I had in my head when we chatted about it, but I *think* we may have mis-communicated on this.

I was more referring to the MessageType field of JsonOutputTemplate as a way to specify what type of output we were doing here.

To automate MessageType, we could probably use a custom json marshaller ([documentation](https://pkg.go.dev/encoding/json#Marshaler)) and use the reflect package to pull the type, and/or rely upon an override in an options bag.